File: InMotionGIT.Common/MetadataCatalogAttributes/ElementSourceAttribute.vb
Namespace Attributes

    ''' <summary>
    ''' Establece la procedencia de la propiedad a nivel de source
    ''' </summary>
    ''' <remarks></remarks>
    <AttributeUsage(AttributeTargets.Property)>
    Public NotInheritable Class ElementSourceAttribute
        Inherits Attribute

        Private _source As String

        Public ReadOnly Property Source() As String
            Get
                Return _source
            End Get
        End Property

        Public Sub New(ByVal source As String)
            _source = source
        End Sub

    End Class

End Namespace

File: Common/InMotionGIT.Common.Domain/Attributes/ElementSourceAttribute.cs
using System;

namespace InMotionGIT.Common.Domain.Attributes;


/// <summary>
/// Establece la procedencia de la propiedad a nivel de source
/// </summary>
/// <remarks></remarks>
[AttributeUsage(AttributeTargets.Property)]
public sealed class ElementSourceAttribute : Attribute
{

    private string _source;

    public string Source
    {
        get
        {
            return _source;
        }
    }

    public ElementSourceAttribute(string source)
    {
        _source = source;
    }

}